### ISSUE TITLE
Modules for pre- and post-processing

### ISSUE THREAD

**Issue by jppelteret** [dealii/dealii#10104]
This relates to @bangerth's [recent comment](https://groups.google.com/d/msgid/dealii/f5d39635-4854-96b6-be11-99ee420754a9%40colostate.edu) on the forum. I had absolutely no idea that this class existed, and could have used it on many occasions. It got me wondering: Should we made two new modules that collect up all of the classes that could be useful for pre- and post-processing?

**Comment by bangerth** [comment]
There is already this: https://dealii.org/developer/doxygen/deal.II/group__IO.html  Maybe it could be keyed off it as a sub-module?

**Comment by jppelteret** [comment]
Yes, that's a good idea and definitely seems like the right place to put it. Its nice that the "Graphical Output" submodule already has the DataPostprocessor[Scalar/Vector/Tensor] classes listed there.
